packages/cli/src/cmd/serve.js

```diff
@@ -0,0 +1,122 @@
+const chokidar = require('chokidar');
+const path = require('path');
+
+const { Site } = require('@markbind/core');
+const { pageVueServerRenderer } = require('@markbind/core/src/Page/PageVueServerRenderer');
+
+const fsUtil = require('@markbind/core/src/utils/fsUtil');
+const { INDEX_MARKDOWN_FILE } = require('@markbind/core/src/Site/constants');
+
+const cliUtil = require('../util/cliUtil');
+const liveServer = require('../lib/live-server');
+const logger = require('../util/logger');
+const {
+  addHandler,
+  changeHandler,
+  generateServerConfig,
+  lazyReloadMiddleware,
+  removeHandler,
+} = require('../util/serveUtil');
+
+function serve(userSpecifiedRoot, options) {
+  if (options.dev) {
+    logger.useDebugConsole();
+  }
+
+  let rootFolder;
+  try {
+    rootFolder = cliUtil.findRootFolder(userSpecifiedRoot, options.siteConfig);
+
+    if (options.forceReload && options.onePage) {
+      logger.error('Oops! You shouldn\'t need to use the --force-reload option with --one-page.');
+      process.exitCode = 1;
+      process.exit();
+    }
+  } catch (error) {
+    logger.error(error.message);
+    process.exitCode = 1;
+  }
+
+  const logsFolder = path.join(rootFolder, '_markbind/logs');
+  const outputFolder = path.join(rootFolder, '_site');
+
+  const presentDefaultFile = fsUtil.fileExists(INDEX_MARKDOWN_FILE) ? INDEX_MARKDOWN_FILE : false;
+  if (options.onePage === true && !presentDefaultFile) {
+    logger.error('Oops! It seems that you didn\'t have the default file index.md.');
+    process.exitCode = 1;
+    process.exit();
+  }
+  let onePagePath = options.onePage === true ? presentDefaultFile : options.onePage;
+  onePagePath = onePagePath ? fsUtil.ensurePosix(onePagePath) : onePagePath;
+
+  const reloadAfterBackgroundBuild = () => {
+    logger.info('All opened pages will be reloaded.');
+    liveServer.reloadActiveTabs();
+  };
+
+  const site = new Site(rootFolder, outputFolder, onePagePath,
+                        options.forceReload, options.siteConfig, options.dev,
+                        options.backgroundBuild, reloadAfterBackgroundBuild);
+
+  // server config
+  const serverConfig = generateServerConfig(options, outputFolder);
```

Comment:
Since this method is only used once, can we revert it back to a constant config? I think it could be clearer as a constant variable that is defined where it is used.

Reply:
Thanks for the good suggestion! Updated in [cb7dc59](https://github.com/MarkBind/markbind/pull/2239/commits/cb7dc59a94894f2c9ab995d975891fdda9d54f7c).